File: Eglantine/Data/Events/LivingRoom_events.lua
static_index = 1

function lookAtPainting()
	if GameState.PaintingOpened then
		Event:ShowMessage("What a waste of a painting...")
	else
		Event:ShowMessage("What a magnificent creature!")
	end
end

function interactWithPainting()
	if Event:UsingItem("Scissors") then
		runCoroutine(function()
			Event:MovePlayerTo("Painting")
			waitUntil("Player stopped")
			--Here, the player does some cool cutting of the painting
			Event:PlaySound("Extend")
			GameState.PaintingOpened = true
			Event:EnableInteractable("Tear")
			Event:EnableInteractable("PaintingDoor")
			Event:DisableInteractable("Painting")
		end)
	end
end

function interactWithTV()
	runCoroutine(function()
		if not GameState.TVOn then
			Event:MovePlayerTo("TV")
			waitUntil("Player stopped")
			--Clicking sound effect
			waitSeconds(1)
			Event:ShowMessage("No good.  It won't turn on.")
		else
			Event:MovePlayerTo("TV")
			waitUntil("Player stopped")
			--Clicking sound effect
			waitSeconds(1)
			Event:ShowMessage("It won't turn off.  The dials don't do anything.")
		end
	end)
end


function lookAtTV()
	if not GameState.TVOn then
		Event:ShowMessage("That TV looks pretty old.")
	else
		Event:OpenTV()
	end
end


function turnOnTV()
	if Event:PlayerHasItem("Photograph") then
		GameState.TVOn = true
		checkTV()
		Event:DisableTrigger("TVActivate")
	end
end


function checkTV()
	if GameState.TVOn then
		setTVImage()
		static_index = 1
		runCoroutine(staticLoop)
		Event:PlaySoundLooping("static", .5, 0, 0)
		--Begin playing static sound, but find a way to turn it off on leave
	end
end

function staticLoop()
	while Event:PlayerInRoom("LivingRoom") do
	
		if Event:PlayerInRoom("LivingRoom") then
			Event:EnableInteractable("Static"..static_index)
		end
		
		waitSeconds(.05)
		
		if Event:PlayerInRoom("LivingRoom") then
			Event:DisableInteractable("Static"..static_index)
		end
		
		static_index = static_index+1
		if static_index > 6 then
			static_index = 1
		end
	end
end



function setTVImage()
	if GameState.KitchenWindowBroken and not GameState.KitchenFlowerPicked then
		Event:SetTVImage("Graphics/TV/tv_kitchen")
		return
	else
		Event:SetTVImage("Graphics/TV/tv_eglantine")
	end	
end

function leaveLivingRoom()
	Event:StopSoundEffect("static")
end

File: Eglantine/Scripts/RoomEvents/LivingRoomEvents.cs
using System;
using System.Collections.Generic;
using ObjectivelyRadical.Scheduler;

namespace Eglantine.Engine
{
	public static partial class GameEvents
	{

		public static IEnumerator<ScriptPauser> lookAtPainting ()
		{
			if (GameState.Instance.PaintingOpened)
			{
				EventManager.Instance.ShowMessage("What a waste of a painting...");
			}
			else
			{
				EventManager.Instance.ShowMessage("What a magnificent creature!");
			}

			yield return null;
		}

		public static IEnumerator<ScriptPauser> interactWithPainting ()
		{
			Console.WriteLine ("Painting time");
			if (EventManager.Instance.UsingItem ("Scissors"))
			{
				EventManager.Instance.MovePlayerTo ("Painting");
				yield return waitUntil ("Player stopped");
					             
				EventManager.Instance.PlaySound ("Extend");
				GameState.Instance.PaintingOpened = true;
				EventManager.Instance.EnableInteractable ("Tear");
				EventManager.Instance.EnableInteractable ("PaintingDoor");
				EventManager.Instance.DisableInteractable ("Painting");
			}
			else
			{
				Console.WriteLine("no hasami");
			}

			yield return null;
		}

		public static IEnumerator<ScriptPauser> interactWithTV ()
		{
			if (!GameState.Instance.TVOn)
			{
				EventManager.Instance.MovePlayerTo("TV");
				yield return waitUntil("Player stopped");
				
				yield return waitSeconds(1);
				EventManager.Instance.ShowMessage("No good.  It won't turn on.");
			}
			else
			{
				EventManager.Instance.MovePlayerTo("TV");
				yield return waitUntil("Player stopped");

				yield return waitSeconds(1);
				EventManager.Instance.ShowMessage("It won't turn off.  The dials don't do anything.");
			}

			yield return null;
		}

		public static IEnumerator<ScriptPauser> lookAtTV ()
		{
			if (!GameState.Instance.TVOn)
			{
				EventManager.Instance.ShowMessage("That TV looks pretty old.");
			}
			else
			{
				EventManager.Instance.OpenTV();
			}

			yield return null;
		}


		public static IEnumerator<ScriptPauser> turnOnTV ()
		{
			if(EventManager.Instance.PlayerHasItem("Photograph"))
			{
				GameState.Instance.TVOn = true;
				checkTV();
				EventManager.Instance.DisableTrigger("TVActivate");
			}

			yield return null;
		}

		public static IEnumerator<ScriptPauser> enterLivingRoom()
		{
			checkTV();

			yield return null;
		}

		public static IEnumerator<ScriptPauser> loadLivingRoom()
		{
			Scheduler.Execute(GameEvents.startIndoorSounds);
			checkTV();
			
			yield return null;
		}

		public static void checkTV ()
		{
			if (GameState.Instance.TVOn)
			{
				setTVImage();
				Scheduler.Execute(staticLoop);
				EventManager.Instance.PlaySoundLooping("static", .5f, 0, 0);
			}
		}

		// Put all logic for changing the image on the screen here
		public static void setTVImage ()
		{
			if (GameState.Instance.KitchenWindowBroken && !GameState.Instance.KitchenFlowerPicked)
			{
				EventManager.Instance.SetTVImage ("Graphics/TV/tv_kitchen");
			}
			else
			{
				EventManager.Instance.SetTVImage ("Graphics/TV/tv_eglantine");
			}
		}


		public static IEnumerator<ScriptPauser> staticLoop ()
		{
			int staticIndex = 1;
			while(EventManager.Instance.PlayerInRoom("LivingRoom"))
			{
				if(EventManager.Instance.PlayerInRoom("LivingRoom")) 
				{
					EventManager.Instance.EnableInteractable("Static" + staticIndex);
				}

				yield return waitSeconds(.05f);

				if(EventManager.Instance.PlayerInRoom("LivingRoom"))
				{
					EventManager.Instance.DisableInteractable("Static" + staticIndex);
				}

				staticIndex++;
				if(staticIndex > 6) staticIndex = 1;
			}
		}


		public static IEnumerator<ScriptPauser> leaveLivingRoom ()
		{
			EventManager.Instance.StopSoundEffect("static");
			yield return null;
		}


// --DOORS-- //
		public static IEnumerator<ScriptPauser> useLivingRoomDoor_Painting()
		{
			door("PaintingDoor", "SecretRoom", "Door");
			yield return null;
		}

		public static IEnumerator<ScriptPauser> useLivingRoomDoor_Foyer()
		{
			door("Door", "Foyer", "LivingRoomDoor");
			yield return waitUntil("Player stopped");
			EventManager.Instance.PlaySound("door", .9f, 0f, 0f);
		}

	}
}

